src/AWS/Orleans.Streaming.SQS/Streams/SQSStreamBuilder.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Orleans.Configuration;
+using Orleans.Hosting;
+using OrleansAWSUtils.Streams;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orleans.Streams
+{
+    public class SiloSqsStreamConfigurator: SiloPersistentStreamConfigurator
+    {
+        public SiloSqsStreamConfigurator(string name, ISiloHostBuilder builder)
+            : base(name, builder)
+        {
+            this.siloBuilder
+                .ConfigureApplicationParts(parts => parts.AddFrameworkPart(typeof(SQSAdapterFactory).Assembly))
+                .AddPersistentStreams(name, SQSAdapterFactory.Create)
+                .ConfigureServices(services =>
+                {
+                    services.ConfigureNamedOptionForLogging<SqsOptions>(name)
+                        .ConfigureNamedOptionForLogging<SimpleQueueCacheOptions>(name)
+                        .ConfigureNamedOptionForLogging<HashRingStreamQueueMapperOptions>(name);
+                });
+        }
+
+        public SiloSqsStreamConfigurator ConfigureAzureQueue(Action<OptionsBuilder<SqsOptions>> configureOptions)
+        {
+            this.Configure<SqsOptions>(configureOptions);
+            return this;
+        }
+        public SiloSqsStreamConfigurator ConfigureCacheSize(int cacheSize = SimpleQueueCacheOptions.DEFAULT_CACHE_SIZE)
+        {
+            this.Configure<SimpleQueueCacheOptions>(ob => ob.Configure(options => options.CacheSize = cacheSize));
+            return this;
+        }
+
+        public SiloSqsStreamConfigurator ConfigureQueueMapper(int numOfQueues = HashRingStreamQueueMapperOptions.DEFAULT_NUM_QUEUES)
+        {
+            this.Configure<HashRingStreamQueueMapperOptions>(ob => ob.Configure(options => options.NumQueues = numOfQueues));
+            return this;
+        }
+    }
+
+    public class ClusterClientSqsStreamConfigurator : ClusterClientPersistentStreamConfigurator
+    {
+        public ClusterClientSqsStreamConfigurator(string name, IClientBuilder builder)
```

Comment:
Client needs to know about the queue count as well (mapper settings), as it uses the mapper to determine which queues to send messages too.  This is probably the case with all stream providers that have configurable numbers of queues (azure, memory, ..)

Reply:
If we're only using the default queue count in all of our tests, we should probably configure a non-default in at least a few tests.  Don't think we need custom tests for this, just not use the defaults in all tests.

Reply:
good point